tests/domain/usecases/slack/test_usecase_post_activity.py

```diff
@@ -0,0 +1,185 @@
+import dataclasses
+import re
+
+import pytest
+
+from slackhealthbot.domain.models.activity import (
+    ActivityData,
+    ActivityHistory,
+    ActivityZone,
+    ActivityZoneMinutes,
+)
+from slackhealthbot.domain.usecases.slack import usecase_post_activity
+
+
+@pytest.mark.parametrize(
+    [
+        "input_value",
+        "expected_output",
+    ],
+    [
+        (0, "➡️"),
+        (1, "➡️"),
+        (-1, "➡️"),
+        (100, "⬆️"),
+        (-100, "⬇️"),
+        (5, "↗️"),
+        (-5, "↘️"),
+    ],
+)
+def test_get_activity_minutes_change_icon(
+    input_value: str,
+    expected_output: str,
+):
+    actual_output = usecase_post_activity.get_activity_minutes_change_icon(input_value)
+    assert actual_output == expected_output
+
+
+@pytest.mark.parametrize(
+    [
+        "input_value",
+        "expected_output",
+    ],
+    [
+        (0, "➡️"),
+        (1, "➡️"),
+        (-1, "➡️"),
+        (100, "⬆️"),
+        (-100, "⬇️"),
+        (35, "↗️"),
+        (-35, "↘️"),
+    ],
+)
+def test_get_activity_calories_change_icon(
+    input_value: str,
+    expected_output: str,
+):
+    actual_output = usecase_post_activity.get_activity_calories_change_icon(input_value)
+    assert actual_output == expected_output
+
+
+@pytest.mark.parametrize(
+    [
+        "input_value",
+        "input_all_time_top_value",
+        "input_recent_top_value",
+        "expected_output",
+    ],
+    [
+        (12, 14, 13, ""),
+        (13, 14, 13, "New record (last 2 days)! 🏆"),
+        (13, 13, 13, "New all-time record! 🏆"),
+    ],
+)
+def test_get_ranking_text(
+    input_value: int,
+    input_all_time_top_value: int,
+    input_recent_top_value: int,
+    expected_output: str,
+):
+    actual_output = usecase_post_activity.get_ranking_text(
+        input_value,
+        input_all_time_top_value,
+        input_recent_top_value,
+        record_history_days=2,
+    )
+    assert actual_output == expected_output
+
+
+@dataclasses.dataclass
+class CreateMessageScenario:
+    name: str
+    activity_history: ActivityHistory
+    expected_message_regex: str
+
+
+CREATE_MESSAGE_SCENARIOS = [
+    CreateMessageScenario(
+        name="new all-time record",
+        activity_history=ActivityHistory(
+            latest_activity_data=ActivityData(
```

Comment:
Maybe we can move part of the scenario setup into the test function (all-time/recent records, and latest activity), and only keep the "new activity" data in the parametrized test scenario.